crates/ra_hir_def/src/nameres/collector.rs

```diff
@@ -908,15 +858,20 @@ where
             return;
         }
 
-        // Case 2: try to resolve in legacy scope and expand macro_rules, triggering
-        // recursive item collection.
+        // Case 2: try to resolve in legacy scope and expand macro_rules
         if let Some(macro_def) = mac.path.as_ident().and_then(|name| {
             self.def_collector.def_map[self.module_id].scope.get_legacy_macro(&name)
         }) {
```

Comment:
Note that I tried to move this "resolve" code inside the `resolve_macro`, however it did not work out. It is because legacy scope macro must be resolved in order. See `plain_macros_are_legacy_textual_scoped`:

https://github.com/rust-analyzer/rust-analyzer/blob/6e10a9f57815ad865a570816436adfdf0de1cdf0/crates/ra_hir_def/src/nameres/tests/macros.rs#L305